pennylane/math/_is_independent.py

```diff
@@ -0,0 +1,317 @@
+    r"""Generate random arguments of a given structure.
+
+    Args:
+        args (tuple): Original input arguments
+        interface (str): Interface of the QNode into which the arguments will be fed
+        num (int): Number of random argument sets to generate
+        seed (int): Seed for random generation
+        bounds (tuple[int]): Range within which to sample the random parameters.
+
+    Returns:
+        list[tuple]: List of length ``num`` with each entry being a random instance
+        of arguments like ``args``.
+
+    This function generates ``num`` many tuples of random arguments in the given range
+    that have the same shapes as ``args``.
+    """
+    width = bounds[1] - bounds[0]
+    if interface == "tf":
+        import tensorflow as tf  # pylint: disable=import-outside-toplevel
+
+        tf.random.set_seed(seed)
+        rnd_args = []
+        for _ in range(num):
+            _args = (tf.random.uniform(tf.shape(_arg)) * width + bounds[0] for _arg in args)
+            _args = tuple(
+                tf.Variable(_arg) if isinstance(arg, tf.Variable) else _arg
+                for _arg, arg in zip(_args, args)
+            )
+            rnd_args.append(_args)
+    elif interface == "torch":
+        import torch  # pylint: disable=import-outside-toplevel
+
+        torch.random.manual_seed(seed)
+        rnd_args = [
+            tuple(torch.rand(np.shape(arg)) * width + bounds[0] for arg in args) for _ in range(num)
+        ]
+    else:
+        np.random.seed(seed)
+        rnd_args = [
+            tuple(np.random.random(np.shape(arg)) * width + bounds[0] for arg in args)
+            for _ in range(num)
+        ]
+
+    return rnd_args
+
+
+def _is_independent_num(func, interface, args, kwargs, test_kwargs):
+    """Test whether a function returns the same output at random positions.
+
+    Args:
+        func (callable): Function to be tested
+        interface (str): Interface used by ``func``
+        args (tuple): Positional arguments with respect to which to test
+        kwargs (dict): Keyword arguments for ``func`` at which to test;
+            The ``kwargs`` are kept fix in this test.
+        test_kwargs (dict): Options for the test.
+
+    Returns:
+        bool: Whether ``func`` returns the same output at the randomly
+        chosen points.
+
+    The available options via ``test_kwargs`` are
+
+      - ``num_pos=5``: Number of random positions to test
+      - ``seed=9123``: Seed for random number generator
+      - ``atol=1e-6``: Absolute precision for comparing the outputs
+      - ``rtol=0``: Absolute precision for comparing the outputs
+      - ``bounds=(-np.pi, np.pi)``: Limits of the range from which to sample
+
+    """
+    test_kwargs = test_kwargs or {}
+    num_pos = test_kwargs.get("num_pos", 5)
+    seed = test_kwargs.get("seed", 9123)
+    atol = test_kwargs.get("atol", 1e-6)
+    rtol = test_kwargs.get("rtol", 0)
+    bounds = test_kwargs.get("bounds", (-np.pi, np.pi))
```

Comment:
Just a sudden thought: what happens if some of the inputs are restricted to certain values, such as integers? e.g.,

```python
def circuit(x, basis):
    qml.BasisState(basis, wires=[0, 1])
    qml.RX(x ** 2, wires=0)
```
I guess it isn't an issue, since we are only checking for linear classical processing, and the gate is never instantiated so no errors will be raised 🤔

Reply:
Will look into this, don't understand it entirely right away...